Sovereign-Labs/sovereign-sdk
# Various improvements to `demo-rollup/README.md`

# Description
This is an attempt at making `examples/demo-rollup/README.md` more pleasing to the eye, simpler to follow, and overall better. Follow-up to #448.

Before: https://github.com/Sovereign-Labs/sovereign-sdk/blob/main/examples/demo-rollup/README.md
After: https://github.com/Sovereign-Labs/sovereign-sdk/blob/filippo/demo-rollup-improvs/examples/demo-rollup/README.md

Most notable changes:

- Discord banner
- Table of Contents generated with https://github.com/thlorenz/doctoc
- I moved the *Getting Started* before the *How to Customize This Example*; I think it's reasonable to assume we want people to first try it out and not have new users look through the README to understand where the basic tutorial starts.
- Widespread changes to formatting, occasional re-wording, fixed some typos. Also tried to make the use of second person more consistent throughout the tutorial but it's not perfect yet (I'll address it during the review based on feedback if we deem it as not good enough).

## Linked Issues
This PR is **not** a solution to https://github.com/Sovereign-Labs/sovereign-sdk/issues/476, which requires some deeper contents changes.

## Testing
After modifying the README, I manually followed the process multiple time, also attempting slight variations and deviations. I believe it's working identical as before, as I haven't really changed the code blocks. Still, it would be great if someone else checked.

I recognize the Table of Contents can be a bit of a pain point, I wouldn't want anybody to forget to update it when modifying the README. We can either:

- Check it's correct on CI (by running `doctoc` and diffing the file). 👈 EDIT: this is what I ended up doing. 
- Make it part of the `stable` release checklist.
- Add a pre-commit hook.

I'm interested in feedback on these options.
